lib/xwax/timecoder.c

```diff
@@ -499,6 +516,57 @@ static void process_bitstream(struct timecoder *tc, signed int m)
 	  tc->valid_counter);
 }
 
+/*
+ * Compare the last quadrant we were in to the new one and return the
+ * correct displacement for the pitch filter model.
+ *
+ * A full revolution of the carrier has the length of 1.0 / tc->def->resolution,
+ * which is also the sample rate of the timecode (not the audio). One quadrant
+ * corresponds to 1/4 * revolution, which is the time between two zero
+ * crossings. Hence we multiply this quantity by displacement in quadrants.
+ */
+
+static double quantize_phase(struct timecoder *tc)
+{
+    unsigned diff = ((tc->quadrant - tc->last_quadrant) % 4);
+    static unsigned long long direction_change_counter = 0;
+
+    /* Check for a displacement of four quadrants */
+    if (diff == 0 && !tc->direction_changed) {
+        return 1.0 / tc->def->resolution;
+
+    /* Check for a displacement of three quadrants */
+    } else if ((tc->forwards && diff == 3) || (!tc->forwards && diff == 1)) {
+        return (3.0 / tc->def->resolution) / 4.0;
+
+    /* Check for a displacement of two quadrants  */
+    } else if (diff == 2) {
+        return (1.0 / tc->def->resolution) / 2.0;
+
+    /* Fallback */
+    } else {
+        return (1.0 / tc->def->resolution) / 4.0;
+    }
```

Comment:
with returns, no need for `else`, it's implied.  and "fallback" is intuitive from the code flow, so we don't need that comment.  Only making this nit because it's easy to produce a suggestion.

```suggestion
    if (diff == 0 && !tc->direction_changed) {
        return 1.0 / tc->def->resolution;
    }
    
    /* Check for a displacement of three quadrants */
    if ((tc->forwards && diff == 3) || (!tc->forwards && diff == 1)) {
        return (3.0 / tc->def->resolution) / 4.0;
    }
    
    /* Check for a displacement of two quadrants  */
    if (diff == 2) {
        return (1.0 / tc->def->resolution) / 2.0;
    }
    
    return (1.0 / tc->def->resolution) / 4.0;
```